src/ucp/tag/offload.c

```diff
@@ -537,7 +537,9 @@ ucs_status_t ucp_tag_offload_sw_rndv(uct_pending_req_t *self)
     rndv_hdr_len = sizeof(ucp_rndv_rts_hdr_t) + ucp_ep_config(ep)->tag.rndv.rkey_size;
     rndv_rts_hdr = ucs_alloca(rndv_hdr_len);
     packed_len   = ucp_tag_rndv_rts_pack(rndv_rts_hdr, req);
-    ucs_assert((rndv_rts_hdr->address != 0) || !UCP_DT_IS_CONTIG(req->send.datatype));
+    ucs_assert((rndv_rts_hdr->address != 0) || !UCP_DT_IS_CONTIG(req->send.datatype) ||
+               !ucp_rndv_is_get_zcopy(req->send.mem_type,
```

Comment:
how is this related to fixing tcp?

Reply:
it fixes another problem reproduced when I added the test (DC/RC related):
https://github.com/openucx/ucx/pull/4579#issue-352777513 (the second item in `WHY?`)
unfortunately, can't split it to another PR as the one test reproduces two problems